File: my_problems/day39-bit-manipulations-1/q3_countSetBits.js
/*****
 * Number of 1 Bits
 * 
 * Problem Description

Write a function that takes an integer and returns the number of 1 bits it has.

Problem Constraints
1 <= A <= 10^9

Input Format
First and only argument contains integer A

Output Format
Return an integer as the answer

Example Input
Input1:
11

Example Output
Output1:
3

Example Explanation
Explanation 1:
11 is represented as 1011 in binary.
 */

const numSetBits = (A) => {

    let count = 0;

    for (let i = 0; i < 32; i++) {
        if ((A >> i) & 1) {
            count++;
        }
    }
    console.log('no of set bits in ', A, ' are ', count);
    return count;
}

numSetBits(15);
numSetBits(13);
numSetBits(64);
numSetBits(10);
numSetBits(255);

File: my_problems/day39-bit-manipulations-1/q3_Count_Set_Bits.java
/* Count set bits */
public class q3_Count_Set_Bits {

    public int numSetBits(int A) {
        int cnt = 0;
        while (A > 0) {
            A = A & (A - 1);
            cnt++;
        }
        System.out.println(cnt);
        return cnt;
    }

    public static void main(String[] args) {
        q3_Count_Set_Bits t1 = new q3_Count_Set_Bits();
        t1.numSetBits(5);
        t1.numSetBits(3);
        t1.numSetBits(103);
        t1.numSetBits(54268);
        t1.numSetBits(16);
    }
}